app/presenters/tree_builder_alert_profile_obj.rb

```diff
@@ -0,0 +1,72 @@
+class TreeBuilderAlertProfileObj < TreeBuilder
+  def initialize(name, type, sandbox, build = true, assign = nil)
+    @assign = {
+      :new     => assign[:new],
+      :current => assign[:current]
+    }
+
+    build = false unless @assign[:new][:assign_to]
```

Comment:
Im not sure, i wrote this before christmas :D But i think, that problem was, i did not wanted the condition in HAML to not render the tree and not call the tree class at all. It was eaiser to have only one condition here. You think its wrong?